app/src/main/java/org/simple/clinic/allpatientsinfacility/AllPatientsInFacilityListAdapter.kt

```diff
@@ -2,19 +2,26 @@ package org.simple.clinic.allpatientsinfacility
 
 import android.view.LayoutInflater
 import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
 import org.simple.clinic.widgets.recyclerview.ViewHolderX
 
-class AllPatientsInFacilityListAdapter : ListAdapter<AllPatientsInFacilityListItem, ViewHolderX>(
-    AllPatientsInFacilityListItem.AllPatientsInFacilityListItemCallback()
-) {
+class AllPatientsInFacilityListAdapter(
+    diffCallback: DiffUtil.ItemCallback<AllPatientsInFacilityListItem>
+) : ListAdapter<AllPatientsInFacilityListItem, ViewHolderX>(diffCallback) {
+
+  private val eventSubject = PublishSubject.create<AllPatientsInFacilityListItem.Event>()
+
+  val listItemEvents: Observable<AllPatientsInFacilityListItem.Event> = eventSubject.hide()
 
   override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderX {
     return ViewHolderX(LayoutInflater.from(parent.context).inflate(viewType, parent, false))
   }
 
   override fun onBindViewHolder(holder: ViewHolderX, position: Int) {
```

Comment:
Why is this named `ViewHolderX`?

Reply:
Just a convention for Kotlin Android Extensions. Package names for kotlin android extensions related code all start with `kotlinx`, and this class is meant to be used for viewholders which need access to synthetic accessors, so `VIewHolderX`.